tools/devops/automation/scripts/GitHub.psm1

```diff
@@ -268,6 +298,138 @@ class GitHubComments {
+        $headers = @{
+            Authorization = ("Bearer {0}" -f $this.Token)
+        }
+
+        $query = @"
+query {
+    repository(owner:"$($this.Org)", name:"$($this.Repo)"){
+        pullRequest(number: $prID) {
+            comments(first: 100) {
+                edges {
+                    node {
+                        id
+                        isMinimized
+                        body
+                        author {
+                            login
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+"@
+        $payload = @{
+            query=$query
+        }
+        $body = ConvertTo-Json $payload
+        $response= Invoke-RestMethod -Uri "https://api.github.com/graphql" -Headers $headers -Method "POST" -Body $body
+        # loop over the result and remove all the extra noise we are not interested in
+        $comments = [System.Collections.ArrayList]@()
+        foreach ($edge in $response.data.repository.pullRequest.comments.edges) {
+            $commentId = $edge.node.id
+            $isMinimized = $edge.node.isMinimized
+            $author = $edge.node.author.login
+            $body = $edge.node.body
+            $comments.Add([GitHubComment]::new($commentId, $author, $body, $isMinimized))
+        }
+        # at this point, we have the comments for the PR, but not the comments of the commits of the PR, yes, confusing. The github UI
+        # contains 2 sets of comments:
+        # 1. Comments on a PR (which is an issue really)
+        # 2. Comments on the commits that are part of the PR
+        #
+        # We are missing those in 2, we can get those with a second query
+
+        $query = @"
+query{
+    repository(owner:"$($this.Org)", name:"$($this.Repo)") {
+        pullRequest(number: $prID){
+            commits(first:100) {
+                edges {
+                    node {
+                        commit {
+                            ... on Commit {
+                                oid
+                                message
+                            } 
+                            comments (first:100) {
+                                edges {
+                                    node {
+                                        id
+                                        body
+                                        author {
+                                            login
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+"@
+        $payload = @{
+            query=$query
+        }
+        $body = ConvertTo-Json $payload
+        $response= Invoke-RestMethod -Uri "https://api.github.com/graphql" -Headers $headers -Method "POST" -Body $body
```

Comment:
I've seen this URL embedded in two different places in the code - should it be in a variable?